utils/kubernetes/kube_state_processor.py

```diff
@@ -143,6 +143,72 @@ def kube_deployment_spec_replicas(self, message, **kwargs):
             self.gauge(metric_name, val, tags)
 
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_requested_cpu_cores(self, message, **kwargs):
+        """ CPU cores requested for a container in a pod. """
+        metric_name = NAMESPACE + '.container.cpu_requested'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            #TODO: add deployment/replicaset?
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_requested_memory_bytes(self, message, **kwargs):
+        """ Memory bytes requested for a container in a pod. """
+        metric_name = NAMESPACE + '.container.memory_requested'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_limits_cpu_cores(self, message, **kwargs):
+        """ CPU cores limit for a container in a pod. """
+        metric_name = NAMESPACE + '.container.cpu_limit'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            #TODO: add deployment/replicaset?
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: namespace, pod, container, node
+    def kube_pod_container_limits_memory_bytes(self, message, **kwargs):
+        """ Memory byte limit for a container in a pod. """
+        metric_name = NAMESPACE + '.container.memory_limit'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # TODO: implement kube_pod_container_status_ready
+    # TODO: implement kube_pod_container_status_running
+    # TODO: implement kube_pod_container_status_terminated
+    # TODO: implement kube_pod_container_status_waiting
+
+    # Labels attached: namespace, pod, container
+    def kube_pod_container_status_restarts(self, message, **kwargs):
+        """ Number of desired pods for a deployment. """
+        metric_name = NAMESPACE + '.container.restarts'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # TODO: implement kube_pod_info
+    # TODO: implement kube_pod_status_ready
+    # TODO: implement kube_pod_status_scheduled
+
+    # Labels attached: namespace, pod, phase=Pending|Running|Succeeded|Failed|Unknown
+    def kube_pod_status_phase(self, message, **kwargs):
```

Comment:
Do you mean like kube_node_status_ready/kube_nod_status_out_of_disk? They live in kube_state_processor.py already, not in kubernetes_state.py.